pkg/apis/acme/v1/types_issuer.go

```diff
@@ -766,6 +766,10 @@ type ACMEIssuerDNS01ProviderRFC2136 struct {
 	// ``HMACSHA1``, ``HMACSHA256`` or ``HMACSHA512``.
 	// +optional
 	TSIGAlgorithm string `json:"tsigAlgorithm,omitempty"`
+
+	// Protocol to use for DNS queries. Valid values are (case-sensitive) ``tcp`` and ``udp``; ``udp`` (default).
```

Comment:
I think the values `TCP` and `UDP` would align better with Kubernetes API best practices. Example (protocol): https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.33/#containerport-v1-core.

Maybe also add an API validation for this using the `+kubebuilder:validation:Enum` marker. Docs: https://book.kubebuilder.io/reference/markers/crd-validation

Reply:
Also @erikgb it also looks like some other fields might benefit from enum approach with a custom string type similar to how its done in `types_challenge.go`. But that's probably out of the scope of this card.. I can file an issue to improve it as a follow up.

Reply:
Adding enum validations to an existing field might be technically breaking. So we should at least not mix it in here. 😊

Please create a follow up issue!

Reply:
yup for sure :)